src/delay_cycles.rs

```diff
@@ -72,6 +72,19 @@ struct Selection {
     remainder: u64,
 }
 
+// Compute the intended number of cycles to delay, and panic if it is greater than
+// the maximum supported amount of cycles
+const fn compute_total_cycles(cycles: u64, mul: u64, div: u64) -> u64 {
+    const MAX_SUPPORTED_CYCLES: u64 = 25_769_803_784;
+    
+    // Multiply first to avoid precision loss, and expand to u128 to avoid overflow
```

Comment:
I wonder how costly the u128 is on our dev machines vs a checked\_mul. At least it sidetracks all the corner cases without having to think of the details.

Reply:
At worst it's like four 64-bit multiplies, or sixteen 32-bit multiplies, and some additions. Compared to the cost of running the rust compiler in the first place, this is nothing.

Reply:
And this is how; with this line of thinking; that our modern car UIs are slow. Our smartphones lags. Or some websites take 5s to display a 140 characters piece of text.

I don't have a benchmark to compare with. I just wanted to point out that it is easy to slow down things by compounding them.

All the code in this library will be executed once per invocation of delay\_cycles... for every single compilation!

Anyways, as I said; I truly don't know the real cost. Just something to keep in mind.